compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -5144,6 +5144,40 @@ abstract class GenJSCode extends plugins.PluginComponent
       (patchedParams, patchedBody)
     }
 
+    // Methods to deal with JSName ---------------------------------------------
+
+    def genPropertyName(name: JSName)(implicit pos: Position): js.PropertyName = {
+      name match {
+        case JSName.Literal(name) => js.StringLiteral(name)
+
+        case JSName.Computed(sym) =>
+          js.ComputedName(genComputedJSName(sym), encodeComputedNameIdentity(sym))
+      }
+    }
+
+    def genExpr(name: JSName)(implicit pos: Position): js.Tree = name match {
+      case JSName.Literal(name) => js.StringLiteral(name)
+      case JSName.Computed(sym) => genComputedJSName(sym)
+    }
+
+    private def genComputedJSName(sym: Symbol)(implicit pos: Position): js.Tree = {
+      /* By construction (i.e. restriction in PrepJSInterop), we know that sym
+       * must be a static method.
+       * Therefore, at this point, we can invoke it by loading its owner and
+       * calling it.
+       */
+      val module = genLoadModule(sym.owner)
+
+      if (isRawJSType(sym.owner.tpe)) {
+        if (!isScalaJSDefinedJSClass(sym.owner) || isExposed(sym))
+          genJSCallGeneric(sym, module, args = Nil, isStat = false)
+        else
+          genApplyJSClassMethod(module, sym, arguments = Nil)
+      } else {
+        genApplyMethod(module, sym, arguments = Nil)
+      }
```

Comment:
I didn't really try that, because things are decomposed early into JS/Scala calls in the `genApply` giant dispatch. So seeing them all together like this never came up before. I think it's OK like that, if a bit ugly.